### PR TITLE
Feature/get assigned jira tickets

### PR DESCRIPTION
**Pull request by baristaGeek**
## Description
We're gonna show the user the Jira tickets that are currently assigned to her. 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  

## Notes
Still gotta figure out how to fix the CORS error.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to the "watermelon" extension will be documented in this fil
 
 Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.
 
+# [1.5.2]
```
Comment (EstebanDalelR) [inline CHANGELOG.md:7]: 1.5.2 is already released

**Comment by EstebanDalelR** [comment]
How will the user login to wm?

File: media/utils/addJiraTicketsToDailySummary.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  //     paintJiraTickets(data);
+  // })
+  // .catch(error => {
+  //     console.log(error);
+  // });
+
+  $("#jiraHolder").append(`<h3>Assigned Jira Tickets</h3>`);
+  jiraTickets.forEach((ticket) => {
+    if (ticket.status === "Done") {
+      $("#jiraHolder").append(`
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply media/utils/addJiraTicketsToDailySummary.js:38]: Ahh yeah! I was gonna ask you this question. Yeah, let's insert it up there.

**Comment by baristaGeek** [comment]
> How will the user login to wm?

Does WM-39 give us the answer?

**Comment by baristaGeek** [comment]
As an additional note: The change in tsconfig.json was required in order for husky to let me commit. Apparently it has something to do with axios (not the node version).

File: media/utils/addJiraTicketsToDailySummary.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        <h3>Assigned Jira Tickets</h3>
+            <div class="Box-header">
+              <h5 class="Box-title Truncate">
+                  ${ticket?.key}
+              </h5>
+              <span style="color: green">${ticket?.fields.status.name}</span>
```
Comment (EstebanDalelR) [inline media/utils/addJiraTicketsToDailySummary.js:11]: Does every ticket have a this field?

File: media/utils/addJiraTicketsToDailySummary.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+              <p>${ticket?.fields.summary}</p>
```
Comment (EstebanDalelR) [inline media/utils/addJiraTicketsToDailySummary.js:12]: Does every ticket have fields?

**Review by EstebanDalelR — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,9 @@ All notable changes to the "watermelon" extension will be documented in this fil
 
 Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.
 
+# [1.6.0]
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply CHANGELOG.md:7]: Absolutely. Let's wait until then to add a version number?

File: media/utils/addJiraTicketsToDailySummary.js (hunk 3 above)
Comment (baristaGeek) [reply media/utils/addJiraTicketsToDailySummary.js:11]: `fields`? Yes

File: src/extension.ts [hunk 6]
```diff
@@ -145,6 +146,15 @@ export async function activate(context: vscode.ExtensionContext) {
       },
     });
 
+    // // call our API to get assigned Jira tickets here
+    const response = await axios.post(
+      "https://app.watermelontools.com/api/jira/getAssignedTicketsInProgress", {
+        // TODO: Make this email programmatic
+        user: "estebandalelr@gmail.com"
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply src/extension.ts:153]: We're waiting for auth to be merged to be able to get this user programmatically, that's why this is a draft PR

File: tsconfig.json [hunk 7]
```diff
@@ -4,7 +4,8 @@
 		"target": "ES2020",
 		"outDir": "out",
 		"lib": [
-			"ES2020"
+			"ES2020",
+			"DOM"
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply tsconfig.json:8]: As I already explained: It's because adding axios forces us to do this. Upgrading the node version (as the logs suggest), don't really allow husky to commit

File: src/extension.ts [hunk 8]
```diff
@@ -145,6 +146,15 @@ export async function activate(context: vscode.ExtensionContext) {
       },
     });
 
+    // // call our API to get assigned Jira tickets here
+    const response = await axios.post(
+      "https://app.watermelontools.com/api/jira/getAssignedTicketsInProgress", {
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply src/extension.ts:151]: For the URL?

File: media/utils/addJiraTicketsToDailySummary.js (hunk 4 above)
Comment (baristaGeek) [reply media/utils/addJiraTicketsToDailySummary.js:12]: yes

File: media/main.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ import setReceivedError from "./utils/setReceivedError.js";
 import removeLoading from "./utils/removeLoading.js";
 import clampCodeBlocks from "./utils/clampCodeBlocks.js";
 import addPRsToDoc from "./utils/addPRsToDoc.js";
+import addJiraTicketsToDailySummary from "./utils/addJiraTicketsToDailySummary.js";
```
[the remark this answers is not in the document]
Comment (baristaGeek) [reply media/main.js:8]: Valid suggestion

File: src/extension.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (EstebanDalelR) [reply src/extension.ts:151]: Yes, there is a precedent of the URL changing

File: media/utils/addJiraTicketsToDailySummary.js (hunk 3 above)
Comment (EstebanDalelR) [reply media/utils/addJiraTicketsToDailySummary.js:11]: Status? Status.name?

File: tsconfig.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (EstebanDalelR) [reply tsconfig.json:8]: Axios was added before july 11. It's already [in this commit](https://github.com/watermelontools/wm-extension/blob/1a78348b12eba8bf1627763c3563cdc3bcae3ef2/package.json) from that date.

[the remark this answers is not in the document]
Comment (EstebanDalelR) [reply tsconfig.json:8]: > As I already explained: It's because adding axios forces us to do this. Upgrading the node version (as the logs suggest), don't really allow husky to commit

So this is not needed in earlier versions of node? When was the requirement introduced?

File: src/extension.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (EstebanDalelR) [reply src/extension.ts:153]: Why not fix it now?
This would allow immediate deploy of 1.6.0

[the remark this answers is not in the document]
Comment (baristaGeek) [reply src/extension.ts:153]: How do I retrieve the email?

File: media/main.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (baristaGeek) [reply media/main.js:8]: Should we release it like this, and fix it on the next version? It's not a core feature.

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,8 +4,8 @@ All notable changes to the "watermelon" extension will be documented in this fil
 
 Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.
 
-## [1.6.0]
-
+# [1.6.0]
```
Comment (EstebanDalelR) [inline CHANGELOG.md:7]: This has already been released

File: CHANGELOG.md [hunk 11]
```diff
@@ -17,6 +17,7 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 
 - Will show the user a prompt to talk to the CTO every third use
 - The hover link will look more like a button
+>>>>>>> feature/authenticator-provider
```
Comment (EstebanDalelR) [inline CHANGELOG.md:20]: What is this?

File: media/main.js [hunk 12]
```diff
@@ -43,7 +44,8 @@ function handleMessage(message) {
       webviewDebugLogger(
         `Received dailySummary: ${JSON.stringify(message.data)}`
       );
-      addDailySummary(message.data);
+      addJiraTicketsToDailySummary(message.data.jiraTickets);
```
Comment (EstebanDalelR) [inline media/main.js:47]: Should this be under Daily Summary?

File: media/utils/addJiraTicketsToDailySummary.js [hunk 13]
```diff
@@ -0,0 +1,29 @@
+const addJiraTicketsToDailySummary = (jiraTickets) => {
+  $("#jiraHolder").append(`<h3>Assigned Jira Tickets</h3>`);
+  jiraTickets.forEach((ticket) => {
+    if (ticket.status === "Done") {
+      $("#jiraHolder").append(`
+        <h3>Assigned Jira Tickets</h3>
```
Comment (EstebanDalelR) [inline media/utils/addJiraTicketsToDailySummary.js:6]: Why the double title?

File: media/utils/addJiraTicketsToDailySummary.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+            <div class="Box-header">
+              <h5 class="Box-title Truncate">
+                  ${ticket?.key}
+              </h5>
+              <span style="color: green">${ticket?.fields?.status?.name}</span>
+              <p>${ticket?.fields?.summary}</p>
+            </div>
+            `);
+    } else {
```
Comment (EstebanDalelR) [inline media/utils/addJiraTicketsToDailySummary.js:15]: The if and else are only different by color, DRY.
Please make the if on the color.

File: package.json [hunk 15]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "watermelon-tools",
   "description": "Watermelon, find all the historical context of your code",
-  "version": "1.6.0",
+  "version": "1.6.1",
```
Comment (EstebanDalelR) [inline package.json:4]: Why is a PR handling this if it's not from `dev` to `main`?

File: src/extension.ts [hunk 16]
```diff
@@ -32,8 +33,9 @@ import multiSelectCommandHandler from "./utils/commands/multiSelect";
 import selectCommandHandler from "./utils/commands/select";
 import debugLogger from "./utils/vscode/debugLogger";
 import checkIfUserStarred from "./utils/github/checkIfUserStarred";
-import { WatermelonAuthenticationProvider } from "./auth";
 import axios from "axios";
```
Comment (EstebanDalelR) [inline src/extension.ts:36]: This is not part of this PR but could we please remove axios, it's unused

File: src/utils/jira/getAssignedJiraTickets.ts [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+  userEmail,
+}: {
+  userEmail: string;
+}) {
+  const assignedJiraTickets = await axios.post(
+    `${backendURL}/api/jira/getAssignedTicketsInProgress`,
+    {
+      userEmail,
+    }
+  ).then(res => res.data);
```
Comment (EstebanDalelR) [inline src/utils/jira/getAssignedJiraTickets.ts:14]: What happens on error?

**Review by EstebanDalelR — CHANGES_REQUESTED** [review]

File: media/main.js (hunk 12 above)
Comment (baristaGeek) [reply media/main.js:47]: How is this a critical error?

File: src/utils/jira/getAssignedJiraTickets.ts (hunk 17 above)
Comment (baristaGeek) [reply src/utils/jira/getAssignedJiraTickets.ts:14]: What do you suggest should happen?

File: media/main.js (hunk 12 above)
Comment (EstebanDalelR) [reply media/main.js:47]: No but prevents debugging down the road.

File: src/utils/jira/getAssignedJiraTickets.ts (hunk 17 above)
Comment (EstebanDalelR) [reply src/utils/jira/getAssignedJiraTickets.ts:14]: Tell the user? Failing silently might be a problem. 
Also, should we get a report?

Comment (baristaGeek) [reply src/utils/jira/getAssignedJiraTickets.ts:14]: Ok

File: media/main.js (hunk 12 above)
Comment (baristaGeek) [reply media/main.js:47]: This is a non-essential feature and doesn't deserve so much attention. We're not even sure if we're gonna keep it/drastically change it

File: src/utils/jira/getAssignedJiraTickets.ts (hunk 17 above)
Comment (baristaGeek) [reply src/utils/jira/getAssignedJiraTickets.ts]: I added a block of code that sends report to the telemetry system. Not sure if this is the way to do it since other utils files don't have such. Does this look reasonable?

File: media/utils/addJiraTicketsToDailySummary.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+const addJiraTicketsToDailySummary = (jiraTickets) => {
+  $("#jiraHolder").append(`<h3>Assigned Jira Tickets</h3>`);
+  jiraTickets.forEach((ticket) => {
+    <div class="Box-header">
+      <h5 class="Box-title Truncate">${ticket?.key}</h5>
+      <span style="color: red">${ticket?.fields?.status?.name}</span>
```
Comment (EstebanDalelR) [inline media/utils/addJiraTicketsToDailySummary.js:6]: We used to have a green version when this was done, why was that removed?

File: src/utils/jira/getAssignedJiraTickets.ts [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+  userEmail: string;
+}) {
+  const assignedJiraTickets = await axios.post(
+    `${backendURL}/api/jira/getAssignedTicketsInProgress`,
+    {
+      userEmail,
+    }
+  ).then(res => res.data).catch(err => {
+    let reporter = analyticsReporter();
+    reporter?.sendTelemetryEvent("error", {
```
Comment (EstebanDalelR) [inline src/utils/jira/getAssignedJiraTickets.ts:17]: Why not send[ an Exception](https://github.com/microsoft/vscode-extension-telemetry#sending-exceptions)?

**Review by EstebanDalelR — CHANGES_REQUESTED** [review]

File: media/utils/addJiraTicketsToDailySummary.js (hunk 18 above)
Comment (EstebanDalelR) [reply media/utils/addJiraTicketsToDailySummary.js:6]: Always red. ok

**Review by EstebanDalelR — APPROVED** [review]
